base/complex.jl

```diff
@@ -56,11 +56,29 @@ Return both the real and imaginary parts of the complex number `z`.
 """
 reim(z) = (real(z), imag(z))
 
-real{T<:Real}(::Type{T}) = T
-real{T<:Real}(::Type{Complex{T}}) = T
+"""
+    real{R<:Real}(::Type{R})
```

Comment:
Do you think that a similar fallback should be defined for `complex` and/or `float` as well?
```
float(T::Type) = typeof(float(zero(T)))
complex(T::Type) = typeof(complex(zero(T)))
```